matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/aggregation/livelocation/LiveLocationAggregationProcessor.kt

```diff
@@ -56,8 +67,39 @@ internal class LiveLocationAggregationProcessor @Inject constructor() {
 
         Timber.d("updating summary of id=$targetEventId with isLive=${content.isLive}")
 
-        aggregatedSummary.endOfLiveTimestampMillis = content.getBestTimestampMillis()?.let { it + (content.timeout ?: 0) }
-        aggregatedSummary.isActive = content.isLive
+        val endOfLiveTimestampMillis = content.getBestTimestampMillis()?.let { it + (content.timeout ?: 0) }
+        aggregatedSummary.endOfLiveTimestampMillis = endOfLiveTimestampMillis
+        aggregatedSummary.isActive = isLive
+
+        if (isLive) {
+            scheduleDeactivationAfterTimeout(targetEventId, roomId, endOfLiveTimestampMillis)
+        } else {
+            cancelDeactivationAfterTimeout(targetEventId, roomId)
+        }
+    }
+
+    private fun scheduleDeactivationAfterTimeout(eventId: String, roomId: String, endOfLiveTimestampMillis: Long?) {
+        endOfLiveTimestampMillis?.let { endOfLiveMillis ->
```

Comment:
Nit: personnally I prefer to have this first line:
```kotlin
endOfLiveTimestampMillis?: return
```
It has the advantage to avoid to "rename" `endOfLiveTimestampMillis` to `endOfLiveMillis` and also reduce the indentation level.

Reply:
Yes I agree it is more readable. I changed it in [cfdc18d](https://github.com/vector-im/element-android/pull/6128/commits/cfdc18d421df0e311b299cf267d2f040e96561ea)